test/functional/editions_controller_test.rb

```diff
@@ -113,4 +113,91 @@ class EditionsControllerTest < ActionController::TestCase
       end
     end
   end
+
+  context "#unpublish" do
+    setup do
+      artefact = FactoryBot.create(
+        :artefact,
+        slug: "test2",
+        kind: "guide",
+        name: "test",
+        owning_app: "publisher",
+      )
+      @guide = GuideEdition.create!(title: "test", slug: "test2", panopticon_id: artefact.id)
+    end
+
+    should "redirect to edition_path when user does not have govuk-editor permission" do
+      user = FactoryBot.create(:user, :welsh_editor, name: "Stub User")
+      login_as(user)
```

Comment:
Could we move it to the setup,
```
  def login_as_welsh_editor
    user = FactoryBot.create(:user, :welsh_editor, name: "Stub User")
    login_as(user)
  end
```
 to avoid repetitions below

Reply:
that would mean we will have to create govuk-editor user for other test, so I am not sure if that would reduce repitations